relays/client-substrate/src/client.rs

```diff
@@ -178,6 +194,24 @@ impl<C: Chain> Client<C> {
 }
 
 impl<C: Chain> Client<C> {
+	/// Return spec version
```

Comment:
I'd suggest to merge two methods - otherwise there's a risk of race when you read twice from non-upgraded and then upgraded chain.

Reply:
+ fields are always used together, if I get it correctly